File: src/Domain/Entities/Appointment.cs
using Domain.Common;
using Domain.Entities.Identity;

namespace Domain.Entities;

public enum AppointmentStatus
{
    Pending = 0,
    Accepted = 1,
    Refused = 2
}

public class Appointment : AuditableAndSoftDeletableEntity
{
    public Guid MemberId { get; set; }
    public Guid AdminId { get; set; }
    public DateTime Date { get; set; }
    public TimeSpan Duration { get; set; } = TimeSpan.FromHours(1);
    public string? Motif { get; set; }
    public AppointmentStatus Status { get; set; } = AppointmentStatus.Pending;
    public string? RefusalReason { get; set; }
    public Guid ConversationId { get; set; }

    public User Member { get; set; } = null!;
    public User Admin { get; set; } = null!;
    public Conversation Conversation { get; set; } = null!;

    public void Accept()
    {
        Status = AppointmentStatus.Accepted;
    }

    public void Refuse(string? reason = null)
    {
        Status = AppointmentStatus.Refused;
        RefusalReason = reason?.Trim();
    }
}

File: src/Web/vue-app/src/types/entities/appointment.ts
export enum AppointmentStatus {
  Pending = 0,
  Accepted = 1,
  Refused = 2
}

export interface Appointment {
  id: string
  memberId: string
  adminId: string
  date: string
  duration: string
  motif: string | null
  status: AppointmentStatus
  refusalReason: string | null
  conversationId: string
}

export interface AvailableSlot {
  date: string
  startTime: string
  endTime: string
}

export interface AvailabilitySlot {
  id?: string
  dayOfWeek: number
  startTime: string
  endTime: string
}

export interface AvailabilityOverride {
  id?: string
  date: string
  startTime: string | null
  endTime: string | null
  isBlocked: boolean
}

export interface AvailabilityData {
  recurring: AvailabilitySlot[]
  overrides: AvailabilityOverride[]
}
